File: Aplicaciones Distribuidas/PARCIAL3/C7_CRUD_VB_GRUPO05/APLICATIVO/CrudLayered_VB/DataAccess/Repositories/EmployeeRepository.vb
Imports System.Data.SqlClient
Imports System.Data
Imports DataAccess

Public Class EmployeeRepository
    Inherits MasterRepository
    Implements IEmployeeRepository

    Private selectAll As String
    Private insert As String
    Private update As String
    Private delete As String

    Public Sub New()
        selectAll = "SELECT * FROM Employee"
        insert = "INSERT INTO Employee VALUES(@idNumber,@name,@mail,@birthday)"
        update = "UPDATE Employee SET IdNumber=@idNumber,Name=@name,Mail=@mail,Birthday=@birthday WHERE idPK=@idPK"
        delete = "DELETE FROM Employee WHERE idPk=@idPK"
    End Sub

    Public Function GetAll() As IEnumerable(Of Employee) Implements IGenericRepository(Of Employee).GetAll
        Dim resultTable = ExecuteReader(selectAll)
        Dim listEmployees = New List(Of Employee)
        For Each item As DataRow In resultTable.Rows
            listEmployees.Add(New Employee() With {
                .IdPk = item(0),
                .IdNumber = item(1),
                .Name = item(2),
                .Mail = item(3),
                .Birthday = item(4)
            })
        Next
        Return listEmployees
    End Function
    Public Function Add(entity As Employee) As Integer Implements IGenericRepository(Of Employee).Add
        parameters = New List(Of SqlParameter)
        parameters.Add(New SqlParameter("@idNumber", entity.IdNumber))
        parameters.Add(New SqlParameter("@name", entity.Name))
        parameters.Add(New SqlParameter("@mail", entity.Mail))
        parameters.Add(New SqlParameter("@birthday", entity.Birthday))
        Return ExecuteNonQuery(insert)
    End Function

    Public Function Edit(entity As Employee) As Integer Implements IGenericRepository(Of Employee).Edit
        parameters = New List(Of SqlParameter)
        parameters.Add(New SqlParameter("@idPK", entity.IdPk))
        parameters.Add(New SqlParameter("@idNumber", entity.IdNumber))
        parameters.Add(New SqlParameter("@name", entity.Name))
        parameters.Add(New SqlParameter("@mail", entity.Mail))
        parameters.Add(New SqlParameter("@birthday", entity.Birthday))
        Return ExecuteNonQuery(update)
    End Function



    Public Function Remove(idPk As Integer) As Integer Implements IGenericRepository(Of Employee).Remove
        parameters = New List(Of SqlParameter)
        parameters.Add(New SqlParameter("@idPK", idPk))
        Return ExecuteNonQuery(delete)
    End Function

    Public Function GetBySalary() As IEnumerable(Of Employee) Implements IEmployeeRepository.GetBySalary
        Throw New NotImplementedException()
    End Function

End Class

File: Aplicaciones Distribuidas/PARCIAL3/C6_CRUD_C#_GRUPO05/APLICATIVO/LayeredApp/DataAccess/Repositories/EmployeeRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using DataAccess.Entities;
using DataAccess.Contracts;
using System.Data.SqlClient;
using System.Data;

namespace DataAccess.Repositories
{
    public class EmployeeRepository : MasterRepository, IEmployeeRepository
    {
        private string selectAll;
        private string insert;
        private string update;
        private string delete;

        public EmployeeRepository()
        {
            selectAll = "SELECT * FROM Employee";
            insert = "INSERT INTO Employee VALUES(@idNumber,@name,@mail,@birthday)";
            update = "UPDATE Employee SET IdNumber=@idNumber,Name=@name,Mail=@mail,Birthday=@birthday WHERE idPK=@idPK";
            delete = "DELETE FROM Employee WHERE idPk=@idPK";
        }

        public int Add(Employee entity)
        {
            parameters = new List<SqlParameter>();
            parameters.Add(new SqlParameter("@idNumber", entity.idNumber));
            parameters.Add(new SqlParameter("@name", entity.name));
            parameters.Add(new SqlParameter("@mail", entity.mail));
            parameters.Add(new SqlParameter("@birthday", entity.birthday));
            return ExecuteNonQuery(insert);
        }

        public int Adit(Employee entity)
        {
            parameters = new List<SqlParameter>();
            parameters.Add(new SqlParameter("@idPK", entity.idPk));
            parameters.Add(new SqlParameter("@idNumber", entity.idNumber));
            parameters.Add(new SqlParameter("@name", entity.name));
            parameters.Add(new SqlParameter("@mail", entity.mail));
            parameters.Add(new SqlParameter("@birthday", entity.birthday));
            return ExecuteNonQuery(update);
        }

        public IEnumerable<Employee> GetAll()
        {
            var tableResult = ExecuteReader(selectAll);
            var listEmployee = new List<Employee>();
            foreach (DataRow item in tableResult.Rows)
            {
                listEmployee.Add(new Employee
                {
                    idPk = Convert.ToInt32(item[0]),
                    idNumber = item[1].ToString(),
                    name = item[2].ToString(),
                    mail = item[3].ToString(),
                    birthday = Convert.ToDateTime(item[4])
                });
            }
            return listEmployee;
        }

        public int Remove(int idPk)
        {
            parameters = new List<SqlParameter>();
            parameters.Add(new SqlParameter("@idPK", idPk));
            return ExecuteNonQuery(delete);
        }
    }
}
